docs/assets.rst

```diff
@@ -0,0 +1,673 @@
+      translated strings in an executable format to allow in-browser translation
+      of UI strings.
+    - :ref:`CKEditor <build-ckeditor>` is packaged with custom plugins for
+      MDN's use cases.
+    - Many :ref:`JavaScript libraries <include-js-libraries>` that are used on
+      MDN are included in Kuma's repository for version control.
+    - :ref:`JavaScript bundles <pipeline-js-bundles>`, assembled by
+      ``django-pipeline``, combine several source files into a
+      single minified JavaScript file.
+    - :ref:`CSS bundles <pipeline-css-bundles>`, assembled by
+      ``django-pipeline`` or ``gulp``, combine several source Sass files into a
+      single minified CSS file.
+* **Collecting** - Built assets are collected to the ``/static`` folder.
+    - Django's :ref:`staticfiles <django-staticfiles>` provides a framework
+      of :ref:`Finder <staticfiles-finder>` and
+      :ref:`Storage <staticfiles-storage>` classes for collecting files.
+    - django-pipeline_ augments ``staticfiles``, to allow compiling, bundling,
+      and minifiying JS and CSS assets in the collection phase.
+    - :ref:`gulp <gulp-alternative>` provides alternative tools to ``staticfiles``
+      for local development of assets.
+* **Serving** - Collected assets are served to visitors in development and production
+    - The :ref:`static tag <static-tag>` returns the URL of assets
+      collected by ``staticfiles``.
+    - The :ref:`statici18n tag <statici18n-tag>` returns the URL of localization
+      JavaScript collected by ``staticfiles``.
+    - The :ref:`pipeline tags <pipeline-tags>` ``javascript`` and
+      ``stylesheet`` return the HTML markup for assets processed and collected
+      by ``django-pipeline``.
+    - :ref:`WhiteNoise <serve-whitenoise>` serves static assets as part of the
+      Django process.
+
+.. _build-locale-files:
+
+Extracting and building locale files
+====================================
+Kuma uses Pontoon_ to translate strings in the user interface, in error
+messages, and in emails. These are stored in the mdn-l10n_ repository,
+and included as a `git submodule`_ at ``locale/``.  See the
+:doc:`localization document </localization>` for more details about locales.
+
+Puente_ extracts strings to the Portable Object Template (``.pot``) files. The
+file ``locale/templates/LC_MESSAGES/django.pot`` contains strings from template
+files and Python code. The file ``javascript.pot`` contains strings from
+JavaScript files. Puente looks for the string parameters of ``gettext``
+functions, such as ``gettext()``, the common alias ``_()``, and ``ngettext()``.
+It also parses longer strings in the template tag ``trans``.
+
+Next the changes are merged into the existing Portable Object (``.po``) files,
+such as ``locale/fr/LC_MESSAGES/django.po``, to add new strings and comment out
+removed strings.
+
+Extracting and merging is done with ``make localeextract``, usually
```

Comment:
If you use ``_()`` or ``_gettext()``, it will be trivial to [configure Puente](https://puente.readthedocs.io/en/latest/installation.html#configure) to extract those strings as well.

Reply:
I'm not sure if this belongs in the doc. I'll describe it, and maybe you'll have a suggestion how to document it?

[Puente](https://github.com/mozilla/puente) is a Python library that handles l10n things for Django projects using Jinja2 templates. It provides the ``extract`` management command. [Management commands](https://docs.djangoproject.com/en/1.11/howto/custom-management-commands/) are discovered in the ``management/commands/`` folder of Django apps, so [puente/management/commands/extract.py](https://github.com/mozilla/puente/blob/master/puente/management/commands/extract.py) is the handler for ``./manage.py extract``.

Like most well-written management commands, the code for the ``extract`` command is mostly focused on setting up command line arguments, extracting configurations, and calling another command. In this case, [extract_command](https://github.com/mozilla/puente/blob/master/puente/commands.py#L67-L129). This allows the function that does the work to be [tested](https://github.com/mozilla/puente/blob/master/tests/test_extract.py) more easily.

The code is a wrapper around the [Babel](http://babel.pocoo.org/en/latest/) library, which is an integrated collection of utilities that assist in internationalizing and localizing Python applications, with an emphasis on web-based applications. Puente uses the [Catalog](http://babel.pocoo.org/en/latest/api/messages/catalog.html#catalogs) to represent a gettext message catalog, and [extract_from_dir](http://babel.pocoo.org/en/latest/api/messages/extract.html#babel.messages.extract.extract_from_dir) to extract strings from files.

Reply:
I think writing it out made it clearer to me how to update the docs.

I've avoided linking to the configuration files, because the links would be out of date when the code is updated. After some thought, I agree links to code would be more useful than making the reader find the code, even if out of date. I'm going to try adding the links with a 🍟 emoji, to signal to the reader (and maintainers) that the link is awesome if fresh, and awful if stale.

Reply:
The french fry emoji 🍟 works, but draws too much attention to itself. I'll just have to trust that people know they might be looking at old code.

Reply:
LOL at the frenchfries! I was making the assumption that the localeextract command was defined in the kuma source itself, which is why I asked for a pointer. Sorry.